include/aspect/postprocess/tracer.h

```diff
@@ -35,36 +38,125 @@ namespace aspect
+         * processing.
+         *
+         * @return A pair of strings that will be printed to the screen after
+         * running the postprocessor in two columns; typically the first
+         * column contains a description of what the data is and the second
+         * contains a numerical value of this data. If there is nothing to
+         * print, simply return two empty strings.
+         */
+        virtual
+        std::pair<std::string,std::string> execute (TableHandler &statistics);
+
+        /**
+         * Save the state of this object.
+         */
+        virtual
+        void save (std::map<std::string, std::string> &status_strings) const;
+
+        /**
+         * Restore the state of the object.
+         */
+        virtual
+        void load (const std::map<std::string, std::string> &status_strings);
+
+        /**
+         * Serialize the contents of this class as far as they are not read
+         * from input parameter files.
+         */
+        template <class Archive>
+        void serialize (Archive &ar, const unsigned int version);
+
+        /**
+         * Declare the parameters this class takes through input files.
+         */
+        static
+        void
+        declare_parameters (ParameterHandler &prm);
+
+        /**
+         * Read the parameters this class declares from the parameter file.
+         */
+        virtual
+        void
+        parse_parameters (ParameterHandler &prm);
+
       private:
         /**
          * The world holding the particles
          */
-        Particle::World<dim, Particle::BaseParticle<dim> >              world;
+        Particle::World<dim>              world;
 
         /**
          * The integrator to use in moving the particles
          */
-        Particle::Integrator::Interface<dim, Particle::BaseParticle<dim> >  *integrator;
+        Particle::Integrator::Interface<dim>  *integrator;
+
+        /**
+         * The interpolator to get particle properties
+         */
+        Particle::Interpolator::Interface<dim>  *interpolator;
 
         /**
          * Pointer to an output object
          */
-        Particle::Output::Interface<dim, Particle::BaseParticle<dim> >      *output;
+        Particle::Output::Interface<dim>      *output;
 
         /**
          * Pointer to a generator object
          */
-        Particle::Generator::Interface<dim, Particle::BaseParticle<dim> >   *generator;
+        Particle::Generator::Interface<dim>   *generator;
 
         /**
-         * Whether this set has been initialized yet or not
+         * Pointer to a properties object
          */
-        bool                            initialized;
+        Particle::Property::Manager<dim>   property_manager;
 
         /**
-         * Number of initial particles to create
+         * Whether this set has been initialized yet or not
          */
-        unsigned int                    n_initial_tracers;
+        bool                            initialized;
```

Comment:
This is the kind of variable that is easy to forget to keep synchronized because I think that it probably stores duplicative information. Presumably, it tracks whether something else has already been initialized. Could you query that other thing to see whether the object as a whole has been initialized already?